makelib/common.mk

```diff
@@ -428,7 +428,12 @@ reviewable:
 # ensure generate target doesn't create a diff
 check-diff: generate
 	@$(INFO) checking that branch is clean
-	@git status --porcelain | grep . && $(FAIL)
```

Comment:
@haarchri what was causing this strategy not to work? I had tested it out locally and it seemed to be effective.

Reply:
at the moment it fails everytime in clean and dirty state - because the check is missing ...

Reply:
@haarchri the issue here is that we are always evaluating to `1` because we are either returning `1` from `git status --porcelain | grep .` (when the branch is clean), or we are returning 0 from `git status --porcelain | grep .` which triggers `&& $(FAIL)` and returns `1` (when the branch is dirty). I fee like a cleaner solution could be:
```
if git status --porcelain | grep . ; then $(FAIL); else $(OK) branch is clean; fi
```

How does that look to you? I could be convinced that calling `git status -s` is more obvious than the `grep .` we are doing.

Reply:
@hasheddan  tested your code snip - yes it is cleaner - added in the MR - thanks

final test looks as expected 

```
20:33:16 [ .. ] checking that branch is clean
20:33:16 [ OK ] branch is clean
```

```
20:36:44 [ .. ] checking that branch is clean
 M Testfile
20:36:44 [FAIL]
make: *** [check-diff] Error 1
```

Reply:
Nice! Thanks @haarchri!